core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelectorFactory.java

```diff
@@ -138,20 +139,26 @@ private NodeMap createNodeMap(Optional<CatalogName> catalogName)
         ImmutableSetMultimap.Builder<HostAddress, InternalNode> byHostAndPort = ImmutableSetMultimap.builder();
         ImmutableSetMultimap.Builder<InetAddress, InternalNode> byHost = ImmutableSetMultimap.builder();
         for (InternalNode node : nodes) {
-            try {
-                byHostAndPort.put(node.getHostAndPort(), node);
-
-                InetAddress host = InetAddress.getByName(node.getInternalUri().getHost());
-                byHost.put(host, node);
-            }
-            catch (UnknownHostException e) {
-                if (inaccessibleNodeLogCache.getIfPresent(node) == null) {
-                    inaccessibleNodeLogCache.put(node, true);
-                    LOG.warn(e, "Unable to resolve host name for node: %s", node);
-                }
-            }
+            byHostAndPort.put(node.getHostAndPort(), node);
```

Comment:
We need a similar change in TopologyAwareNodeSelectorFactory, so I suggest we add the following method to InternalNode (right below `getInternalUri`):
```java
public InetAddress getInternalAddress()
        throws UnknownHostException
{
    Optional<InetAddress> address = tryDecodeHostnameToAddress(internalUri.getHost());
    if (address.isPresent()) {
        return address.get();
    }
    return InetAddress.getByName(internalUri.getHost());
}
```
Then we only change one line in the code... something like simply becomes:
```java
for (InternalNode node : nodes) {
    try {
        byHostAndPort.put(node.getHostAndPort(), node);
        byHost.put(node.getInternalAddress(), node);
    }
    catch (UnknownHostException e) {
        if (inaccessibleNodeLogCache.getIfPresent(node) == null) {
            inaccessibleNodeLogCache.put(node, true);
            LOG.warn(e, "Unable to resolve host name for node: %s", node);
        }
    }
}
```

Reply:
Changes incorporated.